res/controllers/Behringer-Extension-scripts.js

```diff
@@ -1101,6 +1102,13 @@
          */
         register: function(component, shift) {
             this.onRegistry(this.componentRegistry.register, component, shift);
+            if (component.midi) {
+                const id = findComponentId(component.midi);
+                if (!Object.prototype.hasOwnProperty.call(this.inputConnections, id)) {
```

Comment:
whats the difference between this and 
```suggestion
                if (this.inputConnections[id] !== undefined) {
```


I mean I see there is a difference in general, but does it make a concrete difference in this case to simplify like this?

Reply:
It was meant be a bit more `null`- & `undefined`-safe. But it looks like it just crashes _differently_ when `inputConnections` is missing.

Reply:
I think its reasonable to assume you `inputConnections` is a valid object, after all you set it up in the constructor. Also testing only for `undefined` explicitly is fine as long as you ensure you never put `null`s into that object.

Reply:
Just for the record: it needs to be `=== undefined`, not `!== undefined`.